src/plugins/intel_gpu/include/intel_gpu/plugin/plugin.hpp

```diff
@@ -19,6 +19,8 @@ class Plugin : public ov::IPlugin {
     std::string m_default_device_id = "0";
     std::map<std::string, cldnn::device::ptr> m_device_map;
     std::map<std::string, ExecutionConfig> m_configs_map;
+    ov::AnyMap m_compiled_model_format;
+    mutable std::string m_compiled_model_format_device_id;
```

Comment:
it does not look thread-safe to have this variable in plugin, because compilation from parallel threads can override this value.

Reply:
I supposed below line add a lock for it:
https://github.com/openvinotoolkit/openvino/blob/e07fe02cb98f996e1c6b68a1b194264f33d35278/src/inference/src/dev/core_impl.cpp#L756

It seems it cannot lock all compile_model:
https://github.com/openvinotoolkit/openvino/blob/e07fe02cb98f996e1c6b68a1b194264f33d35278/src/inference/src/dev/core_impl.cpp#L765

Reply:
Considered thread-safe, maybe we need pass `config+context` when calling `get_property`, in which to parse out the actual device like `compile_model()` has done.

```
            // need to export network for further import from "cache"
            OV_ITT_SCOPE(FIRST_INFERENCE, ov::itt::domains::LoadTime, "Core::compile_model::Export");
            std::string compiled_model_format;
            if (device_supports_internal_property(plugin, ov::internal::compiled_model_format.name())) {
                ov::AnyMap options;
                if (parsedConfig.count(ov::device::id.name())) {
                    options[ov::device::id.name()] = parsedConfig.at(ov::device::id.name());
                }
                if (context) {
                    options["CONTEXT"] = context;
                }
                compiled_model_format =
                    plugin.get_property(ov::internal::compiled_model_format.name(), options).as<std::string>();
            }
```

If not, even we add lock for `m_compiled_model_format_device_id` in `compiled_model()`, it maybe use another model's value for current model for multi-threads cases.

Reply:
I suppose we don't need to store `m_compiled_model_format_device_id` as plugin field at all

Reply:
Supposed there is only one device driver at the same time point, it is yes.